go/database/mpt/proof.go

```diff
@@ -238,6 +260,30 @@ func (p *proofExtractionVisitor) Visit(node Node, info NodeInfo) VisitResponse {
 	return VisitResponseContinue
 }
 
+// updateChildrenHashesComputeEmbedded updates the hashes of the children of the given node.
+// Furthermore, it updates the embedded status of the children.
+func updateChildrenHashesComputeEmbedded(source NodeSource, node Node, children []NodeReference) error {
+	embeddedChildren := make(map[NodeId]bool, len(children))
+	for _, child := range children {
```

Comment:
I think you could skip refreshing the embedded state of the child if the corresponding hash is not dirty. The dirty flag of the hash should cover both -- the hash and the embedded flag.

Reply:
ok, good idea, updated